sssd-2.9 backport of #6792

alexey-tikhonov opened the pull request.

thalman reviewed (approved):
Thanks for the backport, ACK

alexey-tikhonov commented:
Pushed PR: https://github.com/SSSD/sssd/pull/6881

* `sssd-2-9`
    * 50e2fd2428283cf41a12a3789b146b59306eb8c7 - UTILS: fixed USE_AFTER_FREE
    * f7f9f6e5f7cd5cd80617ca69416e7c28c53487b6 - LDAP: fixed leak of `kprinc`
    * 4d12836767f649cb743e4681f8f86eda18bad754 - LDAP: fixed RESOURCE_LEAK
    * 996affcf6e1bc92998d7cf54a5588463eb22a0c6 - KRB5: fixed RESOURCE_LEAK
    * afbf087d14fd47d385a5f84b994263c999f6226c - KRB5: avoid RESOURCE_LEAK
    * 2ed6aa8d4a1012a645af1264073f5a1a7c9ce332 - KRB5: fix memory leak
    * b9fa1af6ee005c802adfd3fdd8d8525f299487da - KRB5: fix memory leak
    * f745621e839487d9f15e52d7204634a8dd7460e7 - KRB5: avoid FORWARD_NULL
    * e124370f57eb9fec359421991271056c4a52a39e - KRB5: use proper function to deallocate mem
    * e19570ef6113c405c38735448d5b1880fb064e22 - KRB5: avoid another attempt to free 'cc' in 'done:' section if first attempt failed.
